dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,249 @@
+package com.dotmarketing.startup.runonce;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+
+import static com.dotcms.util.CollectionsUtils.*;
+
+/**
+ * This upgrade task will migrate all the Virtual Links that exits on the DB (virtual_link table), if there is one virtual link with a 
+ * site that no longer exists it will be deleted. The old virtual_link fields are migrated like this:
+ * Title -> Title
+ * Url -> This is splited, one is the Site and the other one the Uri
+ * Uri -> Forward To
+ * 
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Jun 8, 2017
+ *
+ */
+public class Task04205MigrateVanityURLToContent extends AbstractJDBCStartupTask{
+	
+	private final String SELECT_VIRTUAL_LINK_QUERY = "select inode, title, url, uri from virtual_link";
+	private final String UPDATE_INODE_QUERY = "update inode set type = 'contentlet' where type = 'virtual_link' ";
+	private final String SELECT_HOSTID_BY_NAME_QUERY = "select identifier from contentlet where text1 = ?";
+	private final String SELECT_HOSTLANG_BY_ID_QUERY = "select language_id from contentlet where identifier = ?";
+	private final String INSERT_IDENTIFIER_QUERY = "insert into identifier(id,parent_path,asset_name,host_inode,asset_type)values (?,?,?,?,?)";
+	private String INSERT_CONTENTLET_QUERY = "insert into contentlet(inode, show_on_menu, sort_order, title, mod_date, mod_user, friendly_name, structure_inode, identifier, language_id,"
+			+ " text1, text2, text3, text4, integer1, integer2, integer3, integer4, integer5, integer6, integer7, integer8, integer9, integer10, integer11, integer12, integer13, integer14,"
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 180 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)